substrate/test-runtime/src/lib.rs

```diff
@@ -119,16 +119,13 @@ impl Slicable for Extrinsic {
 
 impl BlindCheckable for Extrinsic {
 	type Checked = Self;
-	type Address = AccountId;
+	type Error = &'static str;
 
-	fn sender(&self) -> &Self::Address {
-		&self.transfer.from
-	}
-	fn check(self) -> Result<Self, &'static str> {
+	fn check(self) -> Result<Self, (Self, Self::Error)> {
 		if ::runtime_primitives::verify_encoded_lazy(&self.signature, &self.transfer, &self.transfer.from) {
 			Ok(self)
 		} else {
-			Err("bad signature")
+			Err((self, "bad signature"))
```

Comment:
is it ever actually necessary to recover the original `self` after a failed `check`?